server/services/openAIService.js

```diff
@@ -147,85 +151,122 @@ const evaluateAnswer = async (interviewText) => {
 
   const response = await callOpenAIAPI(messages, model, maxTokens);
   const processedResponse = processAPIResponse(response, model);
-  console.log("Processed response:", processedResponse);
 
   return processedResponse.result;
 };
 
-const evaluateTranscriptionForAllQuestions = async (userId, jobId) => {
+const updateOrCreateResult = async (
+  interviewId,
+  questionId,
+  scores,
+  feedback = ""
+) => {
+  let result = await ResultService.findResultByInterviewId(interviewId);
+
+  const questionData = {
+    question_id: questionId,
+    scores: scores,
+    feedback: feedback,
+  };
+
+  if (!result) {
+    await Result.create({
+      interview_id: interviewId,
+      question_scores: [questionData],
+    });
+  } else {
+    result.question_scores.push(questionData);
+    await result.save();
+  }
+};
+
+const evaluateTranscriptionForAllQuestions = async (
```

Comment:
Change the name of the function to one question. As you are evaluating for only one question, not all questions.

Reply:
Ok